apps/src/lib/levelbuilder/code-docs-editor/NewProgrammingExpressionForm.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import RailsAuthenticityToken from '@cdo/apps/lib/util/RailsAuthenticityToken';
+import HelpTip from '@cdo/apps/lib/ui/HelpTip';
+
+export default function NewProgrammingExpressionForm({
+  programmingEnvironmentsForSelect
+}) {
+  return (
+    <form action="/programming_expressions" method="post">
+      <RailsAuthenticityToken />
+      <label>
+        Programming Expression Slug
+        <input name="key" style={styles.inputStyle} required />
+        <HelpTip>
+          <p>
+            The programming expression slug is used in URLs and cannot be
+            updated once set. A slug can only contain lowercase letters, numbers
+            and dashes.
```

Comment:
this help tip specifies lower case, but the regex added in programming_expression.rb looks like it accepts upper case too

Reply:
ah good catch!

Reply:
can you remind me, are we going to generate filenames in github based on these slugs? If so, we might want to take some measures to avoid collisions e.g. between "my-expression" and "My-Expression".

Reply:
Good thought -- we currently serialize all the programming expressions into their own files, such as in https://github.com/code-dot-org/code-dot-org/tree/staging/dashboard/config/programming_expressions/gamelab. I think more logic is needed to ensure these keys are valid, including capitalization like you said, but there also needs to be constraints on where a period can go in the key. Do you mind if I figure this out as a follow-up or do you think it should block this PR?

Reply:
separate PR sounds fine!